development/products-lifecycle/search-index.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Products search index
+weight: 42
+useMermaid: true
+---
+
+# Products search index
+
+In PrestaShop, products are natively indexed for search in database, by keywords. 
+
+When a query is made in a search bar, it is sanitized, splitted in words, and queries are made against `ps_search_word`. 
+
+Then, hits are retrieved from `ps_search_index`, to retrieve matching product ids, and then a weighting and a sort is made to return relevant products. 
```

Comment:
In PrestaShop, product search functionality relies on keyword-based indexing. Each search query entered in the search bar undergoes sanitization and is split into individual words. These words are then matched against the `ps_search_word` table. Matching product IDs are retrieved from the `ps_search_index`, followed by the process of weighting and sorting to deliver the most relevant product results.